### ISSUE TITLE
通过MLX模型引擎部署qwen3 ，在maxkb中调用mcp服务时报错！

### ISSUE THREAD

**Issue by wangqun888** [xorbitsai/inference#3674]
### System Info / 系統信息

mac os
Python 3.12.11
xinference, version 1.7.0.post1
源码安装
Package                       Version
----------------------------- -----------
accelerate                    1.7.0
aiobotocore                   2.7.0
aiofiles                      23.2.1
aiohappyeyeballs              2.6.1
aiohttp                       3.12.13
aioitertools                  0.12.0
aioprometheus                 23.12.0
aiosignal                     1.3.2
altair                        5.5.0
annotated-types               0.7.0
anyio                         4.9.0
async-timeout                 5.0.1
attrdict                      2.0.1
attrs                         25.3.0
audioread                     3.0.1
av                            14.4.0
bcrypt                        4.3.0
beautifulsoup4                4.13.4
bitsandbytes                  0.42.0
botocore                      1.31.64
cbor                          1.0.0
certifi                       2025.6.15
cffi                          1.17.1
charset-normalizer            3.4.2
chattts                       0.2.4
click                         8.1.8
cloudpickle                   3.1.1
colorama                      0.4.6
contourpy                     1.3.2
cryptography                  45.0.4
cycler                        0.12.1
datasets                      3.6.0
decorator                     5.2.1
dill                          0.3.8
distro                        1.9.0
ecdsa                         0.19.1
einops                        0.8.1
einx                          0.3.0
encodec                       0.1.1
f5-tts-mlx                    0.2.6
fastapi                       0.110.3
ffmpy                         0.6.0
filelock                      3.18.0
FlagEmbedding                 1.3.5
fonttools                     4.58.4
frozendict                    2.4.6
frozenlist                    1.7.0
fsspec                        2023.10.0
gguf                          0.17.0
gradio                        4.26.0
gradio_client                 0.15.1
groovy                        0.1.2
h11                           0.16.0
hf-xet                        1.1.4
httpcore                      1.0.9
httpx                         0.28.1
huggingface-hub               0.33.0
idna                          3.10
ijson                         3.4.0
importlib_resources           6.5.2
inscriptis                    2.6.0
ir_datasets                   0.5.10
jieba                         0.42.1
Jinja2                        3.1.6
jiter                         0.10.0
jmespath                      1.0.1
joblib                        1.5.1
jsonschema                    4.24.0
jsonschema-specifications     2025.4.1
kiwisolver                    1.4.8
lazy_loader                   0.4
librosa                       0.11.0
llvmlite                      0.44.0
lxml                          5.4.0
lz4                           4.4.4
markdown-it-py                3.0.0
MarkupSafe                    2.1.5
matplotlib                    3.10.3
mdurl                         0.1.2
mlx                           0.26.1
mlx-lm                        0.25.2
mlx-vlm                       0.1.27
mlx-whisper                   0.4.2
modelscope                    1.27.0
more-itertools                10.7.0
mpmath                        1.3.0
msgpack                       1.1.1
multidict                     6.5.0
multiprocess                  0.70.16
narwhals                      1.43.0
networkx                      3.5
numba                         0.61.2
numpy                         1.26.4
nvidia-ml-py                  12.575.51
openai                        1.88.0
opencv-contrib-python         4.11.0.86
opencv-python                 4.10.0.84
optimum                       1.26.1
orjson                        3.10.18
packaging                     25.0
pandas                        2.3.0
passlib                       1.7.4
peft                          0.15.2
pillow                        10.4.0
pip                           25.1.1
platformdirs                  4.3.8
pooch                         1.8.2
propcache                     0.3.2
protobuf                      6.31.1
psutil                        7.0.0
pyarrow                       20.0.0
pyasn1                        0.6.1
pybase16384                   0.3.8
pycparser                     2.22
pydantic                      2.11.7
pydantic_core                 2.33.2
pydub                         0.25.1
Pygments                      2.19.1
pynvml                        12.0.0
pyparsing                     3.2.3
pypinyin                      0.54.0
python-dateutil               2.9.0.post0
python-jose                   3.5.0
python-multipart              0.0.20
pytz                          2025.2
PyYAML                        6.0.2
quantile-python               1.1
qwen-omni-utils               0.0.8
qwen-vl-utils                 0.0.11
referencing                   0.36.2
regex                         2024.11.6
requests                      2.32.4
rich                          13.9.4
rpds-py                       0.25.1
rsa                           4.9.1
ruff                          0.12.0
s3fs                          2023.10.0
safehttpx                     0.1.6
safetensors                   0.5.3
scikit-learn                  1.7.0
scipy                         1.15.3
semantic-version              2.10.0
sentence-transformers         4.1.0
sentencepiece                 0.2.0
setproctitle                  1.3.6
setuptools                    80.9.0
shellingham                   1.5.4
six                           1.17.0
sniffio                       1.3.1
sounddevice                   0.5.2
soundfile                     0.13.1
soupsieve                     2.7
soxr                          0.5.0.post1
sse-starlette                 2.3.6
starlette                     0.37.2
sympy                         1.14.0
tabulate                      0.9.0
tblib                         3.1.0
threadpoolctl                 3.6.0
tiktoken                      0.9.0
timm                          1.0.15
tokenizers                    0.21.1
tomli                         2.2.1
tomlkit                       0.12.0
torch                         2.7.1
torchaudio                    2.7.1
torchvision                   0.22.1
tqdm                          4.67.1
transformers                  4.52.4
transformers-stream-generator 0.0.5
trec-car-tools                2.6
typer                         0.11.1
typing_extensions             4.14.0
typing-inspection             0.4.1
tzdata                        2025.2
unlzw3                        0.2.3
urllib3                       2.0.7
uvicorn                       0.34.3
uvloop                        0.21.0
vector-quantize-pytorch       1.22.17
vocos                         0.1.0
vocos-mlx                     0.0.7
warc3-wet                     0.2.5
warc3-wet-clueweb09           0.2.5
websockets                    11.0.3
wheel                         0.45.1
wrapt                         1.17.2
xinference                    1.7.0.post1
xllamacpp                     0.1.20
xoscar                        0.7.9
xxhash                        3.5.0
yarl                          1.20.1
zlib-state                    0.1.9

### Running Xinference with Docker? / 是否使用 Docker 运行 Xinfernece？

- [ ] docker / docker
- [ ] pip install / 通过 pip install 安装
- [ ] installation from source / 从源码安装

### Version info / 版本信息

xinference, version 1.7.0.post1

### The command used to start Xinference / 用以启动 xinference 的命令

xinference-local --host 0.0.0.0 --port 9997

### Reproduction / 复现过程

通过xinference接入qwen3模型后，无法通过AI对话组件调用mcp服务！

应用中添加mcp服务
<img width="705" alt="Image" src="https://github.com/user-attachments/assets/62e5f092-13c8-4a53-9867-2e7cddfb9386" />
调试直接报错
<img width="497" alt="Image" src="https://github.com/user-attachments/assets/7db02d33-1bc1-4cce-beb8-45caac0a78f2" />

### Expected behavior / 期待表现

帮忙看看是什么问题，谢谢

**Comment by qinxuye** [comment]
看 Xinference 服务日志。

**Comment by wangqun888** [comment]
> 看Xinference服务日志。

日志没有任何输出。

**Comment by qinxuye** [comment]
那请求确定打到 Xinference 了吗？或者要先确认 MaxKB 正确处理了。

**Comment by wangqun888** [comment]
> 那请求确定打到 Xinference 了吗？或者要先确认 MaxKB 正确处理了。

关闭mcp服务，日志有输出。

<img width="1232" alt="Image" src="https://github.com/user-attachments/assets/eb928545-9729-4724-b501-35c26675bd05" />

**Comment by qinxuye** [comment]
那如果是 Xinference 报错，应该在 Xinference 服务有日志的。

**Comment by wangqun888** [comment]
> 那如果是 Xinference 报错，应该在 Xinference 服务有日志的。
奇怪的就是没有日志，无处下手找问题。
maxkb
我也提交了

<img width="1053" alt="Image" src="https://github.com/user-attachments/assets/1cf298b2-95b3-425f-b374-5a580245bb78" />

**Comment by qinxuye** [comment]
接的哪个 MCP，咋操作的方便列出来我们看下怎么复现。

**Comment by wangqun888** [comment]
切换到ollama
<img width="382" alt="Image" src="https://github.com/user-attachments/assets/58bdf272-d9cc-4e9b-91ca-41a7fd2b1c17" />
还是这个mcp服务

<img width="560" alt="Image" src="https://github.com/user-attachments/assets/3f2e16b4-22c8-4f6a-94c4-d33a26c5fcc4" />

正常回复

<img width="813" alt="Image" src="https://github.com/user-attachments/assets/8ab70172-810e-4485-9291-ae27059d3293" />

**Comment by qinxuye** [comment]
给点有用信息好吗

**Comment by qinxuye** [comment]
看起来和 #3675 一样，那个能给出相对有效信息。这里关闭。
